workflows/cloudify_system_workflows/snapshot.py

```diff
@@ -44,17 +43,21 @@ def chunks(g):
             yield ch
 
     s = ''
+    n = 0
     decoder = json.JSONDecoder()
     for ch in chunks(f):
         s += ch
         try:
             while s:
                 obj, idx = decoder.raw_decode(s)
+                n += 1
                 yield json.dumps(obj)
                 s = s[idx:]
         except:
             pass
 
+    assert not n or not s
```

Comment:
Instead of assert we should probably raise some reasonable exception. Like NonRecoverableError for instance..